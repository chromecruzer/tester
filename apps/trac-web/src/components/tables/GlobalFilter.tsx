import 'regenerator-runtime/runtime';
import {useState} from "react";

export const GlobalFilter = ({filteredText, setFilteredText}) => {
  // const [text, setText] = useState(filteredText);
  const handleChange = e => {
    const v = e.target.value;
    console.log(`global filter changed value ${v}`);
    // setText(v);
    setFilteredText(v || null);
  }
  return <span>
    Search:
    <input
      className='bg-bl-consumer-light text-bl-text-grey p-2 '
      placeholder='Search'
      value={filteredText || ''}
      onChange={handleChange}
    />
  </span>
}
