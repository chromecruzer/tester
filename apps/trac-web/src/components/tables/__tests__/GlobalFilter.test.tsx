import {render, screen} from "@testing-library/react";
import {GlobalFilter} from "../GlobalFilter";
import userEvent from "@testing-library/user-event";

describe('GlobalFilter', () => {
  const renderGlobalFilterComponent = (filteredText, setFilteredText) => {
    render(<GlobalFilter
      filteredText={filteredText}
      setFilteredText={setFilteredText}
    />)
  }
  it('should have no value if filtered text is null', () => {
    renderGlobalFilterComponent(null, jest.fn());
    // screen.debug();
    const textbox = screen.queryByRole('textbox');
    // screen.debug(textbox);
    if (textbox === null) {
      fail('Cannot find text box');
    }
    expect(textbox.getAttribute('class')).toMatchSnapshot();
    expect(textbox.getAttribute('placeHolder')).toEqual('Search');
    expect(textbox.getAttribute('value')).toEqual('');
  });
  it('should create a search input with a prefilled value', () => {
    const expectedValue = 'expectedValue'
    renderGlobalFilterComponent(expectedValue, jest.fn());
    // screen.debug();
    const textbox = screen.queryByRole('textbox');
    // screen.debug(textbox);
    if (textbox === null) {
      fail('Cannot find text box');
    }
    expect(textbox.getAttribute('value')).toEqual(expectedValue);

  });
  it('should call setFiltered text twice when a search is entered', () => {
    const expectedFilterText = 'Text to be Filtered';
    const filteredTextSpy = jest.fn();
    renderGlobalFilterComponent(null, filteredTextSpy);
    // screen.debug();
    const textbox = screen.queryByRole('textbox');
    // screen.debug(textbox);
    if (textbox === null) {
      throw new Error('Cannot find text box');
    }
    return userEvent.type(textbox, expectedFilterText).then(() => {
      expect(filteredTextSpy).toBeCalledTimes(expectedFilterText.length);
    })

  });
});
