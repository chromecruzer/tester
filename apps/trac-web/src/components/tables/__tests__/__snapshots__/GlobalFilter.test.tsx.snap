// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFilter should have no value if filtered text is null 1`] = `"bg-bl-consumer-light text-bl-text-grey p-2 "`;
